0xProject/0x-monorepo
# abi-gen should regenerate contract-wrappers if template files change

## Description

Currently, `abi-gen` only checks the artifact file to determine if an existing generated wrapper is outdated. This PR adds a check for template files as well.  Closes #1874 

## Testing instructions

Added a unit test to `abi-gen`. 

You can also try editing artifact/template files and running `yarn generate_contract_wrappers` within `abi-gen-wrappers` to see if new wrappers get generated appropriately.

## Types of changes



* Bug fix (non-breaking change which fixes an issue)





## Checklist:
